utils/sampler/uniform_resample.go

```diff
@@ -70,7 +64,7 @@ func (s *uniformResample) Next() (uint64, error) {
 	}
 
 	for {
-		draw := uint64(s.rng.Int63n(int64(s.length)))
+		draw := s.rng.Uint64n(s.length - 1)
```

Comment:
Note the `-1` here now because `Uint64n` returns values in the range `[0, n]` whereas `Int63n` returned values in the range `[0, n)`